spring-cloud-aws-parameter-store-config/src/main/java/org/springframework/cloud/aws/paramstore/AwsParamStorePropertySourceLocator.java

```diff
@@ -92,8 +92,6 @@ public AwsParamStorePropertySourceLocator(AWSSimpleSystemsManagement ssmClient,
 			addProfiles(this.contexts, baseContext, profiles);
 		}
 
-		Collections.reverse(this.contexts);
```

Comment:
There's been a missing test that justifies this line. Context have to ordered in a way that the most specific ones are first - profiles, application, then the default context. Would be nice if you could include such test and then I think it should fail without reversing the `context` collection

Reply:
Makes sense, will implement test for both Locator classes.

Reply:
@maciejwalkowiak since TreeSet wouldn't allow order of profiles,application and then the default context because String comparing I implemented it with LinkedHashSet to preserve order and remove duplicates.